src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { IconCheckBoxCheck } from '../Icons';
+
+import classNames from 'classnames';
+import styles from './Checkbox.module.scss';
+const cx = classNames.bind(styles);
+
+export interface CheckboxProps {
```

Comment:
Types need adjusting slightly - shouldn't be possible to pass a `link` and not an `onLinkClick`.

Reply:
Updated as link props

Reply:
I think it needs a lil bit of adjusting, something like
```ts
interface CheckboxProps {
    // all but link, onClick
}

interface ButtonCheckboxProps extends CheckboxProps {
    onClick?: () => void;
}

interface LinkCheckboxProps extends CheckboxProps {
    link: string;
    openInNewTab?: boolean; // this should probably default to true
}

export const Checkbox = ({ ... }: CheckboxProps | ButtonCheckboxProps | LinkCheckboxProps) => {
```

Not sure if that's the exact syntax, but the idea is that if `onClick` is defined, Typescript should throw errors if `link` or `openInNewTab` are defined, and vice versa.

Reply:
Updated types